### ISSUE TITLE
Jupyter lab extensions not working

### ISSUE THREAD

**Issue by gioarma** [jupyterlab/jupyterlab#8122]
## Description

I installed jupyter lab extensions, which are enabled, but do not work. They appear in the extension manager, but still then they do not do anything (e.g. the toc extension does not produce the expected toc side panel).
I tried to run `jupyter lab build` but nothing changed.

I saw this issue has been cited also in  krassowski/jupyterlab-lsp#203

By running in the terminal
 `$ jupyter labextension list`
I get:
```
JupyterLab v2.0.1
Known labextensions:
   app dir: /opt/anaconda3/share/jupyter/lab
        @jupyter-widgets/jupyterlab-manager v2.0.0  enabled  OK
        @jupyterlab/toc v3.0.0  enabled  OK
        @lckr/jupyterlab_variableinspector v0.5.0  enabled  OK
        @pyviz/jupyterlab_pyviz v1.0.0  enabled  OK

Uninstalled core extensions:
    @jupyter-widgets/jupyterlab-manager
    @jupyterlab/toc
    @lckr/jupyterlab_variableinspector
    @pyviz/jupyterlab_pyviz

```
Is it normal that these extensions look like they are installed, but are also listed in the uninstalled core extensions?
I did try to uninstall and reinstall them, so I don't know if it's due to this.

## Installed packages
`$ conda list`

```
jupyter_client            6.1.0                      py_0    anaconda
jupyter_contrib_core      0.3.3                      py_2    conda-forge
jupyter_contrib_nbextensions 0.5.1                    py37_0    conda-forge
jupyter_core              4.6.1                    py37_0    anaconda
jupyter_highlight_selected_word 0.2.0                 py37_1000    conda-forge
jupyter_latex_envs        1.4.4                 py37_1000    conda-forge
jupyter_nbextensions_configurator 0.4.1                    py37_0    conda-forge
jupyterlab                2.0.1                      py_0    conda-forge
jupyterlab_server         1.0.7                      py_0    conda-forge
...
nodejs                    13.10.1              h38d8c5a_0    conda-forge
```

**Comment by krassowski** [comment]
It was also mentioned for other extensions by multiple users here: https://discourse.jupyter.org/t/extension-shows-up-as-both-installed-and-uninstalled-and-doesnt-work/2688/4 and I encountered this myself in one of my environments (I _think_ that it was after upgrading JupyterLab).

@gioarma the workaround with removing the `jupyter\lab\settings\build_config.json` file worked for me (it is described on discourse). Did you try it?

**Comment by meeseeksmachine** [comment]
This issue has been mentioned on **Jupyter Community Forum**. There might be relevant details there:

https://discourse.jupyter.org/t/extension-shows-up-as-both-installed-and-uninstalled-and-doesnt-work/2688/5

**Comment by gioarma** [comment]
@krassowski Yes it worked! Thanks!

I also think this was after upgrading Jupyter Lab (not sure anyway)

**Comment by jasongrout** [comment]
Great!

**Comment by jasongrout** [comment]
In second thought, the above printed message about extensions that are installed and uninstalled is really confusing, so leaving open so that can be investigated.

**Comment by RaulCPena** [comment]
I am having the same issue with multiple extensions and will try deleting the jupyter\lab\settings\build_config.json file

**Comment by sebacea** [comment]
same problem here (git extension not working):

```
(base) instr1@F4Linux26:~$ jupyter labextension list
JupyterLab v1.2.6
Known labextensions:
   app dir: /home/instr1/anaconda3/share/jupyter/lab
        @jupyterlab/git v0.10.0  enabled  OK
        @jupyterlab/latex v1.0.0  enabled  OK
        nbdime-jupyterlab v1.0.0  enabled  OK

Uninstalled core extensions:
    @jupyterlab/git
    @jupyterlab/latex
```

**Comment by JTebeka** [comment]
it works 👍 
Thx

**Comment by Shrey8** [comment]
Worked for me as well, thank you!

**Comment by ianhi** [comment]
If you are in a base conda environment the location of the file is a bit different than: `<conda_root>/envs/<env_name>/share/jupyter/lab/settings/build_config.json`

Instead it is in: `<conda_root>/share/jupyter/lab/settings/build_config.json`

**Comment by lasribeiro** [comment]
> It was also mentioned for other extensions by multiple users here: https://discourse.jupyter.org/t/extension-shows-up-as-both-installed-and-uninstalled-and-doesnt-work/2688/4 and I encountered this myself in one of my environments (I _think_ that it was after upgrading JupyterLab).
> 
> @gioarma the workaround with removing the `jupyter\lab\settings\build_config.json` file worked for me (it is described on discourse). Did you try it?

Thank you! This workaround worked for me.

**Comment by peiweiliu** [comment]
weird for no `setting` folder after the path `<conda_root>/share/jupyter/lab/`. so i cannot try to troubleshooting by delete this specific file.:(
<img width="339" alt="Screen Shot 2020-07-31 at 3 30 30 AM" src="https://user-images.githubusercontent.com/43423821/89011465-63f82180-d2de-11ea-958a-b874f054271e.png">
Appreciated for any other possible solution thinking!

**Comment by StephieLaPugh** [comment]
Similar to peiweilu, I have a settings folder, but there's nothing in it to delete (set to show hidden). Any ideas? I really want to use JL, but this feels like a lot of fiddling for a feature that is native to other IDEs.

**Comment by dominijk** [comment]
On mac on 2.1.5 deleting `build_config.json` worked for me

**Comment by LawrentChen** [comment]
Deleting `build_config.json` in `~/opt/anaconda3/share/jupyter/lab/settings` works for me on macos as well. It took me quite a lot of time to find this solution, hope it can be fixed.

**Comment by zZer0o** [comment]
Deleting `build_config.json` works on jupyter lab 2.2.6. And my OS is Arch with kernel 5.8.

I encountered this problem when I had finished installing and uninstalling some extensions. But I'm not sure how to reproduce the issue.

**Comment by ThamuMnyulwa** [comment]
> weird for no `setting` folder after the path `<conda_root>/share/jupyter/lab/`. so i cannot try to troubleshooting by delete this specific file.:(
> <img alt="Screen Shot 2020-07-31 at 3 30 30 AM" width="339" src="https://user-images.githubusercontent.com/43423821/89011465-63f82180-d2de-11ea-958a-b874f054271e.png">
> Appreciated for any other possible solution thinking!

 I have the same issue. No settings file. Perhaps its because we did not install it using anaconda? Where else can one find the build_config.json file ?

**Comment by alexander-manley** [comment]
Yes, confirming that this is still broken as of Jupyterlab version 3.0.12.  This temporary fix should to be in the docs.  

Reminder: This issue was opened over a year ago on Mar 29, 2020. This is Git version control, and not just some random extension.  

Tagging this issue as "build system" may be causing this issue not getting the critical attention that is warranted.

**Comment by krassowski** [comment]
@alexander-manley, I agree that this is a pain point and deserves attention.

> This is Git version control, and not just some random extension.

I just wonder, did you encounter this installing git using `pip install`, or using `jupyter labextension install`? I think that it should not be a problem any longer when using the pip install option (which is suggested for JupyterLab 3, here: https://github.com/jupyterlab/jupyterlab-git/issues/779#issuecomment-763465541). If you however installed with pip install and still see this issue then that would be _interesting_; I wonder, could it be an issue with updating from JupyterLab 2.x?

**Comment by alexander-manley** [comment]
@krassowski, 
Yes, a simple copy & paste of the "pip install" prompt was used. 
The error crops up both on a fresh spin-up of the official Jupyterlab all-spark-notebook container, and even more concerning is that it shows up in the the latest annual April 6 GA build of IntelliJ/PyCharm that ships with Jupyter. 
---
"Failed to load the jupyterlab-git server extension
Git server extension is unavailable. Please ensure you have installed the JupyterLab Git server extension by running: pip install --upgrade jupyterlab-git. To confirm that the server extension is installed, run: jupyter serverextension list."
---

**Comment by gi3265** [comment]
i have \jupyter\lab\settings folder but do not have build_config.json... Any suggestion where i can find it instead??

**Comment by g6ai** [comment]
> i have \jupyter\lab\settings folder but do not have build_config.json... Any suggestion where i can find it instead??

Try uninstall a jupyterlab extension (from web gui or command line) and it will be created.

**Comment by xuoutput** [comment]
> @alexander-manley, I agree that this is a pain point and deserves attention.
> 
> > This is Git version control, and not just some random extension.
> 
> I just wonder, did you encounter this installing git using `pip install`, or using `jupyter labextension install`? I think that it should not be a problem any longer when using the pip install option (which is suggested for JupyterLab 3, here: [jupyterlab/jupyterlab-git#779 (comment)](https://github.com/jupyterlab/jupyterlab-git/issues/779#issuecomment-763465541)). If you however installed with pip install and still see this issue then that would be _interesting_; I wonder, could it be an issue with updating from JupyterLab 2.x?

---

`conda list | grep lab`
<img width="707" alt="image" src="https://user-images.githubusercontent.com/16385443/156990115-7c579414-50bf-4e7f-953e-bfdd746a0278.png">


`pip list | grep lab`
<img width="400" alt="image" src="https://user-images.githubusercontent.com/16385443/156989903-999a81e5-4e28-463f-bbe1-d3a65cf496d6.png">

`jupyter labextension list`
<img width="628" alt="image" src="https://user-images.githubusercontent.com/16385443/156989938-46049565-aa4d-4c84-9b73-bf9b3a4ada59.png">

`ls /root/miniconda3/envs/ext-3/share/jupyter/lab`
<img width="231" alt="image" src="https://user-images.githubusercontent.com/16385443/156990278-e1a002e9-c0ac-4ff3-94dc-41cbd29dca84.png">

**After I uninstall 0.1.0, then I install 0.1.1
why? still 0.1.0 ?**
@krassowski

**Comment by xuoutput** [comment]
> > @alexander-manley, I agree that this is a pain point and deserves attention.
> > > This is Git version control, and not just some random extension.
> > 
> > 
> > I just wonder, did you encounter this installing git using `pip install`, or using `jupyter labextension install`? I think that it should not be a problem any longer when using the pip install option (which is suggested for JupyterLab 3, here: [jupyterlab/jupyterlab-git#779 (comment)](https://github.com/jupyterlab/jupyterlab-git/issues/779#issuecomment-763465541)). If you however installed with pip install and still see this issue then that would be _interesting_; I wonder, could it be an issue with updating from JupyterLab 2.x?
> 
> `conda list | grep lab` <img alt="image" width="707" src="https://user-images.githubusercontent.com/16385443/156990115-7c579414-50bf-4e7f-953e-bfdd746a0278.png">
> 
> `pip list | grep lab` <img alt="image" width="400" src="https://user-images.githubusercontent.com/16385443/156989903-999a81e5-4e28-463f-bbe1-d3a65cf496d6.png">
> 
> `jupyter labextension list` <img alt="image" width="628" src="https://user-images.githubusercontent.com/16385443/156989938-46049565-aa4d-4c84-9b73-bf9b3a4ada59.png">
> 
> `ls /root/miniconda3/envs/ext-3/share/jupyter/lab` <img alt="image" width="231" src="https://user-images.githubusercontent.com/16385443/156990278-e1a002e9-c0ac-4ff3-94dc-41cbd29dca84.png">
> 
> **After I uninstall 0.1.0, then I install 0.1.1 why? still 0.1.0 ?** @krassowski

I delete `yarn.lock` then `jlpm install`, then `python -m build`. 
<img width="851" alt="image" src="https://user-images.githubusercontent.com/16385443/156995140-3c796861-9675-4b9c-a5eb-e59b045af10d.png">
It's ok
<img width="562" alt="image" src="https://user-images.githubusercontent.com/16385443/156995291-494755ad-900c-411c-ba15-9d43de320104.png">

**Comment by jd-santos** [comment]
Same issue. Also on a Mac, not installed through conda, and no settings file under lab. I'm just trying to get my toc extension working.

**Comment by chookity-pokk** [comment]
I am on a MacBook Pro on OS update version 14.0 and going into `<conda_root>/envs/<env_name>/share/` doesn't have a `jupyter` folder and running a search for a `build_config.json` shows up nothing.
